### PR TITLE
feat(context): add FusionContextSearchError

### PR DESCRIPTION
**Pull request by asbjornhaland**
## Why

closes:

### Check off the following:
- [x] Confirm that I checked changes to branch which I am merging into.
  - _I have validated included files_
  - _My code does not generate new linting warnings_
  - _My PR is not a duplicate, [check existing pr`s](https://github.com/equinor/fusion-framework/pulls)_
- [x] Confirm that the I have completed the [self-review checklist](https://github.com/equinor/fusion-framework/blob/main/contributing/self-review.md).

- [x] Confirm that my changes meet our [code of conduct](https://github.com/equinor/fusion-framework/blob/main/CODE_OF_CONDUCT.md).

#### Conditional
- [ ] Confirm project selected and category, actual, iteration are set
- [ ] Confirm Milestone selected _(if any)_

### CODE REVIEW

**Review by odinr — DISMISSED** [review]

**Comment by asbjornhaland** [comment]
Ref AB#46886
Ref AB#44769

File: .changeset/early-ghosts-remain.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+"@equinor/fusion-framework-react-module-context": minor
+"@equinor/fusion-framework-module-context": minor
+---
+
+Add FusionContextSearchError
```
Comment (odinr) [inline .changeset/early-ghosts-remain.md:6]: Add note about possible breaking change of `packages/modules/context/src/ContextProvider.ts`

**Review by odinr — CHANGES_REQUESTED** [review]
🌊 comment

File: .changeset/early-ghosts-remain.md (hunk 1 above)
Comment (asbjornhaland) [reply .changeset/early-ghosts-remain.md]: Updated

**Review by odinr — APPROVED** [review]

**Review by odinr — APPROVED** [review]

**Pull request merged**